torch/utils/data/dataset.py

```diff
@@ -73,6 +73,9 @@ def __len__(self):
         return self.cumulative_sizes[-1]
 
     def __getitem__(self, idx):
+        if idx < 0:
+            assert -idx <= len(self), "absolute value of index should not exceed dataset length"
```

Comment:
It would be slightly better to raise `ValueError` instead of an `AssertionError` here.

Reply:
Done.